docs/source/contributing.rst

```diff
@@ -286,20 +286,31 @@ Code formatting
 ~~~~~~~~~~~~~~~
 
 Instead of wasting time arguing about code formatting, we use `black
-<https://github.com/psf/black>`__ to automatically format all our
-code to a standard style. While you're editing code you can be as
-sloppy as you like about whitespace; and then before you commit, just
-run::
+<https://github.com/psf/black>`__ as well as other tools to automatically
+format all our code to a standard style. While you're editing code you
+can be as sloppy as you like about whitespace; and then before you commit,
+just run::
 
-    pip install -U black
-    black setup.py trio
+    pip install -U pre-commit
+    pre-commit
 
 to fix it up. (And don't worry if you forget – when you submit a pull
 request then we'll automatically check and remind you.) Hopefully this
 will let you focus on more important style issues like choosing good
 names, writing useful comments, and making sure your docstrings are
 nicely formatted. (black doesn't reformat comments or docstrings.)
 
+If you would like, you can even have pre-commit run before you commit by
+running::
+
+    pre-commit install
+
+and now pre-commit will run before git commits. You can uninstall the
+pre-commit hook at any time by running::
+
+    pre-commit uninstall
+
+
 Very occasionally, you'll want to override black formatting. To do so,
 you can can add ``# fmt: off`` and ``# fmt: on`` comments.
```

Comment:
Mind sneaking in how to disable `isort` here too?

Reply:
I don't mind, I can do that.

Reply:
Added in 766c981